honeybadgermpc/batch_reconstruction.py

```diff
@@ -118,51 +150,59 @@ def list_transpose(lst):
 
     Communication takes place over two rounds,
-      objects sent/received of the form ('R1', shares) or ('R2', shares)
+      objects sent/received of the form('R1', shares) or ('R2', shares)
       up to one of each for each party
 
     Reconstruction takes places in chunks of t+1 values
     """
+    bench_logger = logging.LoggerAdapter(logging.getLogger("benchmark_logger"),
+                                         {"node_id": myid})
 
-    fp = GF(p)
-    secret_shares = [v.value for v in secret_shares]
-    round1_chunks = to_chunks(secret_shares, t + 1)
-    num_chunks = len(round1_chunks)
+    # (optional) Induce faults
 
+    secret_shares = [v.value for v in secret_shares]
     if config is not None and config.induce_faults:
         logging.debug("[FAULT][BatchReconstruction] Sending random shares.")
         secret_shares = [random.randint(0, p - 1) for _ in range(len(secret_shares))]
 
-    point = EvalPoint(fp, n, use_fft=use_fft)
-    bench_logger = logging.LoggerAdapter(logging.getLogger("benchmark_logger"),
-                                         {"node_id": myid})
+    # Prepare recv loops for this batch reconstruction
+
     subscribe_task, subscribe = subscribe_recv(recv)
     del recv  # ILC enforces this in type system, no duplication of reads
 
-    task_r1, q_r1 = recv_each_party(subscribe('R1'), n)
-    task_r2, q_r2 = recv_each_party(subscribe('R2'), n)
-    data_r1 = [asyncio.create_task(recv()) for recv in q_r1]
-    data_r2 = [asyncio.create_task(recv()) for recv in q_r2]
+    task_r1, recvs_r1 = recv_each_party(subscribe('R1'), n)
+    data_r1 = [asyncio.create_task(recv()) for recv in recvs_r1]
+
+    task_r2, recvs_r2 = recv_each_party(subscribe('R2'), n)
+    data_r2 = [asyncio.create_task(recv()) for recv in recvs_r2]
     del subscribe  # ILC should determine we can garbage collect after this
 
+    # Set up encoding and decoding algorithms
+    fp = GF(p)
+    decoding_algorithm = Algorithm.GAO if config is None else config.decoding_algorithm
+
+    point = EvalPoint(fp, n, use_fft=use_fft)
     enc = EncoderFactory.get(point, Algorithm.FFT if use_fft else Algorithm.VANDERMONDE)
     dec = DecoderFactory.get(point, Algorithm.FFT if use_fft else Algorithm.VANDERMONDE)
-    decoding_algorithm = Algorithm.GAO
-    if config is not None:
-        decoding_algorithm = config.decoding_algorithm
     robust_dec = RobustDecoderFactory.get(t, point, algorithm=decoding_algorithm)
 
-    # Step 1: Compute the polynomial, then send
+    # Prepare data for step 1
+    round1_chunks = chunk_data(secret_shares, t + 1)
+    num_chunks = len(round1_chunks)
+
+    # Step 1: Compute the polynomial P1, then send the elements
     start_time = time.time()
+
     encoded = enc.encode(round1_chunks)
-    to_send = list_transpose(encoded)
-    for j in range(n):
-        send(j, ('R1', to_send[j]))
+    to_send = transpose_lists(encoded)
+    [send(idx, ('R1', elem)) for idx, elem in enumerate(to_send)]
```

Comment:
I am not sure what's the convention to have list comprehensions without having their values assigned to a variable, can you check? A loop will actually not create a list in the memory which will eventually need to be garbage collected.

Reply:
I looked it up, and you're right-- this isn't pythonic.